### ISSUE TITLE
About the optimizer and scheduler

### ISSUE THREAD

**Issue by Pay20Y** [YongWookHa/swin-transformer-ocr#5]
There seems to be no code call the function "CustomCosineAnnealingWarmupRestarts"? I wonder if the optimizer and scheduler you used are AdamW and Cosine Annealing?
Thanks!

**Comment by YongWookHa** [comment]
The `CustomCosineAnnealingWarmupRestarts` Code is in `utils.py`.

https://github.com/YongWookHa/swin-transformer-ocr/blob/1970e8d003336a7e2b7b41150a27d53b9f7d85a6/utils.py#L44

Have a look at it, plz :)

**Comment by Pay20Y** [comment]
Thanks for you quick reply. My question is that there seems to be no code use the function "CustomCosineAnnealingWarmupRestarts", are there some special usages in the pytorch_lightning?

**Comment by YongWookHa** [comment]
Oh, I guess I misunderstood your question.

https://github.com/YongWookHa/swin-transformer-ocr/blob/1970e8d003336a7e2b7b41150a27d53b9f7d85a6/models.py#L57

Here you can find the usage of `CustomCosineAnnealingWarmupRestarts `.

It depends on what you wrote in your `yaml` file which contains the settings.

https://github.com/YongWookHa/swin-transformer-ocr/blob/1970e8d003336a7e2b7b41150a27d53b9f7d85a6/settings/default.yaml#L14

As you can check, `default.yaml` says it will use `CustomCosineAnnealingWarmupRestarts `.

![image](https://user-images.githubusercontent.com/12293076/150529050-dc2547e1-5187-4783-8321-b0b4754b91d9.png)

This image shows how the learning rate moves with `CustomCosineAnnealingWarmupRestarts `.

I wrote a tech post in [my blog](https://yongwookha.github.io/MachineLearning/2021-10-06-cosine-annealing-warm-up-restarts). Though it's in Korean language, you can use browser translation with chrome or edge:)

Hope this helps you.

**Comment by Pay20Y** [comment]
I got it, thanks again!
